templates/pages/admin/form/form_editor.html.twig

```diff
@@ -0,0 +1,278 @@
+            {# Add action #}
+            <button
+                class="btn btn-primary"
+                type="submit"
+                name="add"
+                form="form-form"
+                title="{{ __("Add") }}"
+            >
+                <i class="ti ti-plus me-1"></i>
+                <span class="d-block add-label">{{ __("Add") }}</span>
+            </button>
+        {% elseif item.canUpdate() %}
+            {# Update action #}
+            <button
+                class="btn btn-primary"
+                type="submit"
+                name="update"
+                form="form-form"
+                title="{{ __("Save") }}"
+            >
+                <i class="ti ti-device-floppy me-1"></i>
+                <span class="d-block save-label">{{ __("Save") }}</span>
+            </button>
+        {% endif %}
+        </button>
+    </div>
+
+    {# Form id #}
+    <input type="hidden" name="id" value="{{ item.fields.id }}">
+
+    {% if item.isNewItem() %}
+        {# CSRF #}
+        <input type="hidden" name="_glpi_csrf_token" value="{{ csrf_token() }}" />
+    {% endif %}
+
+    {{ call_plugin_hook(constant('Glpi\\Plugin\\Hooks::POST_ITEM_FORM'), {'item': item, 'options': params}) }}
+</form>
+
+{# Preview form modal #}
+<div
+    id="preview_form_modal"
+    class="modal modal-blur fade"
+    aria-modal="true"
+    role="dialog"
+    aria-labelledby="preview_form_modal_label"
+>
+    <div class="modal-dialog modal-lg modal-dialog-centered" role="document">
+        <div class="modal-content"></div>
+    </div>
+</div>
+
+<script>
+    // Isolate variables in a self calling function
+    (function(){
+        // Adjust height using javascript
+        // This is the only reliable way to make our content use the remaining
+        // height of the page as the tabs container doesn't define a height
+        const adjust_container_height = function() {
+            const window_height = document.body.offsetHeight ;
+            const editor_height = $('.form-editor-container').offset().top;
+            const tab_content_border = 1; // Border added at the bottom of the page, must be taken into account
+            $('.form-editor-container').css('height', (window_height - editor_height - tab_content_border) + "px");
+        };
+        adjust_container_height();
+
+        // Trigger on resize
+        const adjust_container_height_throttled = _.throttle(adjust_container_height, 100);
+        $(window).on('resize', function() {
+           adjust_container_height_throttled();
+        });
+
+        // Handle preview
+        $('#form-form button[name=preview]').on('click', function(e) {
+            // Read form's id
+            const form = $(e.target).closest('form');
+            const id = form.find('input[name=id]').val();
+
+            // Unsaved changes wont be taken into account as the ajax endpoint will load data from DB using the given id
+            // TODO: trigger a save before preview ?
```

Comment:
The best would probably to pass the "unsaved changes" to the render controller. People may want to preview their changes before saving them. It is easier than having to rollback when the preview shows an unexpected result to the admin.

Reply:
This TODO was already part of the backlog, I've added your comment in its description: https://github.com/orgs/glpi-project/projects/7/views/1?pane=issue&itemId=50762489